R/xml_attr.R

```diff
@@ -95,3 +97,73 @@ xml_attrs.xml_node <- function(x, ns = character()) {
 xml_attrs.xml_nodeset <- function(x, ns = character()) {
   lapply(x, xml_attrs, ns = ns)
 }
+
+#' @param value character vector of new value.
+#' @rdname xml_attr
+#' @export
+`xml_attr<-` <- function(x, attr, ns = character(), value) UseMethod("xml_attr<-")
+
+#' @export
+`xml_attr<-.xml_node` <- function(x, attr, ns = character(), value) {
+  if (is.null(value)) {
+    node_remove_attr(x$node, name = attr, nsMap = ns)
+  } else {
+    node_set_attr(x$node, name = attr, nsMap = ns, value)
+  }
+  x
+}
+
+#' @export
+`xml_attr<-.xml_nodeset` <- function(x, attr, ns = character(), value) {
+  lapply(x, `xml_attr<-`, attr = attr, ns = ns, value = value)
+  x
+}
+
+#' @rdname xml_attr
+#' @export
+`xml_attrs<-` <- function(x, ns = character(), value) UseMethod("xml_attrs<-")
+
+#' @export
+`xml_attrs<-.xml_node` <- function(x, ns = character(), value) {
+  if (!is_named_or_null(value)) {
+    stop("`value` must be a named character vector or `NULL`", call. = FALSE)
+  }
+
+  attrs <- names(value)
```

Comment:
Need to check this is not NULL, and elements not "" or NA? (You can just throw an error if not)

```R
has_names <- function(x) {
  nms <- names(x)
  if (is.null(nms)) {
    rep_along(x, FALSE)
  } else {
    !(is.na(nms) | nms == "")
  }
}
```